src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -632,14 +643,63 @@ trait Namers extends MethodSynthesis {
         )
       }
 
-      sym setInfo {
-        mkTypeCompleter(copyDef) { sym =>
-          assignParamTypes()
-          lazyType complete sym
-        }
+      mkTypeCompleter(copyDef) { sym =>
+        assignParamTypes()
+        lazyType complete sym
```

Comment:
Turns out I already had a commit for that in my wip branch :-) Pushed to this one, plus some more boy scouting.